client/src/controllers/TagController.test.js

```diff
@@ -25,52 +25,156 @@ describe('TagController', () => {
   beforeEach(() => {
     element = null;
     jest.clearAllMocks();
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        json: () => Promise.resolve([]),
+      }),
+    );
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
   });
 
   it('should attach the jQuery tagit to the controlled element', async () => {
+    jest.useRealTimers(); // Switch to real timers for this test
```

Comment:
I tried using `await jest.resolveAllTimersAsync()`, but it doesn't handle `requestAnimationFrame` properly since it relies on real browser timing. Jest's mocked timers like `jest.runAllTimers()` or `jest.resolveAllTimersAsync()` only simulate time-based operations like `setTimeout` and `setInterval`, which don't cover `requestAnimationFrame`. This test depends on real timing for browser-like interactions, so `jest.useRealTimers()` is necessary. However, feel free to double-check if there's an alternative I might have missed.

Reply:
Hmm ok. I'll take another look. Thanks for explaining.